analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,415 @@
+  # Given a data.frame with the percentage values for each region and the
+  # dominant status for that region, plot the dominant status call on the
+  # x-axis with the percentage values on the y-axis.
+  #
+  # Args:
+  #   status_count_df: data.frame with percentage values for each unique status
+  #                    in each unique region (arm/cytoband/gene) and the
+  #                    dominant status for that region
+  #   region_variable: string of the region (which will also be a column name)
+  #                    that the data.frame holds percentage values for
+  # Return:
+  #   status_plot: plot representing the dominant status call on the x-axis and
+  #                the percentage values on the y-axis
+  
+  status_count_df %>%
+    select(-total) %>%
+    dplyr::ungroup() %>%
+    tibble::column_to_rownames(region_variable) %>%
+    tidyr::gather(status, percent,-dominant_status) %>%
+    ggplot2::ggplot(ggplot2::aes(x = status,
+                                 y = percent)) +
+    ggplot2::geom_jitter() +
+    ggplot2::facet_wrap( ~ dominant_status)
+}
+```
+
+
+### Files and directories
+
+```{r}
+results_dir <- "results"
+
+# Define a logical object for running in CI
+running_in_ci <- params$is_ci
+```
+
+### Read in files
+
+Read in cytoband status file and format it for what we will need in this notebook. 
+
+```{r}
+# Read in the file with consensus CN status data and the UCSC cytoband data --
+# generated in `03-add-cytoband-status-consensus.Rmd`
+consensus_seg_cytoband_status_df <-
+  read_tsv(file.path("results", "consensus_seg_with_ucsc_cytoband_status.tsv.gz")) %>%
+  # Need this to not have `chr`
+  mutate(chr = gsub("chr", "", chr),
+         cytoband = paste0(chr, cytoband)) %>%
+  select(
+    chromosome_arm,
+    # Distinguish this dominant status that is based on cytobands, from the status 
+    dominant_cytoband_status = dominant_status,
+    cytoband,
+    Kids_First_Biospecimen_ID
+  )
+```
+
+Read in the chromosome-level and gene-level data. 
+
+```{r}
+# Read in the annotated CN file (without the UCSC data)
+consensus_seg_autosomes_df <-
+  read_tsv(file.path(results_dir, "consensus_seg_annotated_cn_autosomes.tsv.gz"))
+```
+
+Joining the gene-level, cytoband-level, and arm-level data into one data frame.
+
+```{r}
+combined_status_df <- consensus_seg_autosomes_df %>%
+  inner_join(
+    consensus_seg_cytoband_status_df,
+    by = c("biospecimen_id" = "Kids_First_Biospecimen_ID", "cytoband")
+  )
+```
+
+## Define most focal units
+
+### Determine chromosome arm status
+
+```{r}
+# Use our custom function to make the status calls
+arm_status_count <- status_majority_caller(
+  combined_status_df,
+  "chromosome_arm",
+  "status", 
+  percent_threshold = percent_threshold
+)
+
+# Display table
+arm_status_count %>%
+  group_by(dominant_status)
+```
+
+These are the chromosome arms that have not been defined as gain or loss -- we want to define their cytoband/gene level status
```

Comment:
```suggestion
These are the chromosome arms that have not been defined as gain or loss -- we want to define their cytoband/gene-level status
```
Figure we should be consistent with the `-` (This was probably me who left this)